### ISSUE TITLE
[FR] Ghost completions ala Copilot

### ISSUE THREAD

**Issue by NightMachinery** [s-kostyaev/ellama#107]
`ellama-code-complete` inserts the llm output directly into the buffer which is not very ideal. Copilot's "ghost completions" that overlay a short gray output that you can confirm later on is more useful. https://github.com/zerolfx/copilot.el does this, but it only supports Microsoft Copilot.

Also, there should be a way to limit the completions to at most N lines and M characters.

**Comment by s-kostyaev** [comment]
Agreed. And be able to manage context properly. Try one of tabby integrations for now, it is in very long term plan. Or maybe [help wanted]

**Comment by NightMachinery** [comment]
@s-kostyaev Thanks. What models does tabby support? Skimming its docs, it seems to be a proprietary third-party LLM reseller?

**Comment by s-kostyaev** [comment]
I prefer to use deepseek coder 6.7b with tabby - it has very good performance for my use cases.

> Skimming its docs, it seems to be a proprietary third-party LLM reseller?

No. It pulls models from hf and run it locally.

**Comment by s-kostyaev** [comment]
See https://tabby.tabbyml.com/

**Comment by abcdw** [comment]
One of the possible solutions can be using https://code.bsdgeek.org/adam/corfu-candidate-overlay with usual capf.

**Comment by tusharhero** [comment]
> One of the possible solutions can be using
> https://code.bsdgeek.org/adam/corfu-candidate-overlay with usual
> capf.

Emacs 30 has inbuilt `completion-preview-mode` too.

**Comment by tusharhero** [comment]
https://yhetil.org/emacs-devel/m2r0357a5r.fsf@milanglacier.com/

**Comment by tusharhero** [comment]
> https://yhetil.org/emacs-devel/m2r0357a5r.fsf@milanglacier.com/

The project is at [milanglacier/minuet-ai.el](https://github.com/milanglacier/minuet-ai.el)

**Comment by tusharhero** [comment]
> > Actually I feel it should be easy to ignore current session, or
> > use different session for that specific command. Ellama is not
> > very transparent about that.
> 
> I don't understand what are you talking about

See #255 and #283.

Basically, I get constantly surprised by Ellama. I have no clue what
session it is using, I cannot specify the session for the command
easily, nor can I see which session is being used, nor can I ask it to
use a "ephemeral" session specifically for that command alone.
